utensor_cgen/operators.py

```diff
@@ -13,107 +13,115 @@ def snippet(self):
 
 
 class _AddOperator(_Operator):
-  def __init__(self, op_info, **kwargs):
+  def __init__(self, op_info, ref_counts, to_eval):
```

Comment:
You could have left this as `**kwargs` as it will expand to `ref_counts` and `to_eval`

Reply:
I know.
But then I have to do something in the function body like this:
```
ref_counts = kwargs.get('ref_counts')
if ref_counts is None:
  raise ValueError(.....)
```

If you set them as positional arguments, it's required and checked by python interpreter.
I just want to make things clear

Reply:
Pretty sure there is python built ins for doing this cleanly. I think `**kwargs` is more manageable in the long run

Reply:
What is it exactly?
I can't agree with you based on your memory only.

Reply:
Well If you just do `kwargs["ref_counts"]` and reference counts is not in the dictionary then it will throw a `KeyError`

Reply:
But a `KeyError` does not reflect the semantics here.
It's a **required** for the function to work properly.

Just think about the error message:
```
....
KeyError: 'ref_counts'
```
against
```
....
TypeError: __ini__ missing 1 required positional argument: 'ref_counts'
```

To me, the later is much much better than the former.